src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -260,8 +260,10 @@ private void SelectFastest(NodeSelectorState state, int index)
         {
             state.Fastest = index;
             Interlocked.Exchange(ref state.SpeedTestMode, 0);
-
-            _updateFastestNodeTimer.Change(TimeSpan.FromMinutes(1), Timeout.InfiniteTimeSpan);
+            if (_updateFastestNodeTimer == null)
```

Comment:
Let's create a method `EnsureFastestNodeTimerExists()` which initializes the timer if it is null. Use it also in the `ScheduleSpeedTest`. You can execute 'timer.Change' here anyway after ensuring, it is not that heavy.

Reply:
done